File: SotCoreWrapper/CrewService.cpp
#include "pch.h"
#include "CrewService.h"
namespace SoT
{
	CrewService::CrewService(UE4Actor^ actor) : UE4Actor(actor)
	{
		if (!singleton)
			singleton = this;
	}
		
	ACrewService CrewService::getActor()
	{
		return *reinterpret_cast<ACrewService*>(&UE4Actor::getActor());
	}

	array<Crew^>^ CrewService::getCrews()
	{
		TArray<FCrew> crews = getActor().GetCrews();
		array<Crew^>^ CrewsArray = gcnew array<Crew^>(crews.Length());
		for (int i = 0; i < crews.Length(); i++)
		{
			for (int d = 0; d < crews[i].GetPlayers().Length(); d++)
			{
				System::String^ name = gcnew System::String(crews[i].GetPlayers()[d].GetName().c_str());
				d = d;
			}
			CrewsArray[i] = gcnew Crew(crews[i]);
		}
		return CrewsArray;
	}
}

File: SoTCoreExternal/Game/Service/CrewService.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SoT.Game.Service
{
    public class CrewService : UE4Actor
    {
        private TArray Crews
        {
            get
            {
                TArray Crews = SotCore.Instance.Memory.ReadProcessMemory<TArray>(Address + SotCore.Instance.Offsets["ACrewService.Crews"]);
                return Crews;
            }
        }

        public new uint Num
        {
            get
            {
                return (uint)Crews.NumElements;
            }
        }

        public CrewService(UE4Actor actor) : base(actor.Address)
        {
        }

        public CrewService(ulong address) : base(address)
        {
        }

        public Crew GetCrewByIndex(int index)
        {
            return Crews.GetValue<Crew>(index, (int)SotCore.Instance.Offsets["Crew.Size"]);
        }
    }
}